client/fingerprint/env_aws.go

```diff
@@ -46,6 +46,353 @@ var ec2InstanceSpeedMap = map[*regexp.Regexp]int{
 	regexp.MustCompile(`.*\.32xlarge`): 10000,
 }
 
+type ec2Specs struct {
+	mhz   float64
+	cores int
+	model string
+}
+
+func (e ec2Specs) ticks() int {
+	return int(e.mhz) * e.cores
+}
+
+func specs(ghz float64, vCores int, model string) ec2Specs {
+	return ec2Specs{
+		mhz:   ghz * 1000,
+		cores: vCores,
+		model: model,
+	}
+}
+
+// Map of instance type to documented CPU speed.
```

Comment:
wonder if this could be automatically build from open source data like [ec2instances.info](https://ec2instances.info/) or AWS's own [API](https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instance-types.html)

Reply:
Ideally, yes!! Unfortunately ec2instances.info doesn't report `processor-info.sustained-clock-speed-in-ghz` from the API, and using the API requires configuring AWS keys which we don't want Nomad Clients to be burdened with.
Created https://github.com/hashicorp/nomad/issues/7830 which notes an idea of how to automate getting the data from the API